documentation/design-docs/data-catalog.md

```diff
@@ -0,0 +1,497 @@
+  making pull requests against the `data-architecture/` repo.
+* Team members will build views manually by running `dbt run` from
+  the root of the folder.
+* In rare cases where we want to preserve the behavior of old views, we will
+  enable [dbt
+  versioning](https://docs.getdbt.com/docs/collaborate/govern/model-versions)
+  for those views.
+
+#### Building and updating documentation
+
+* Like views, documentation will be edited via pull request.
+* Documentation will be rebuilt by running `dbt docs generate` from the
+  root of the folder.
+* Documentation will be hosted as a static site on S3 with a CloudFront CDN.
+* We will define a helper script in the repo to update documentation.
+* As an optional improvement, we may decide to incorporate docs generation and
+  upload into a CI/CD step (GitHub Action).
+
+#### Running data integrity checks
+
+* Data integrity checks will run via the `dbt test` command, and will be defined
+  as a GitHub Action workflow.
+* Data integrity checks will be run once per night after the sqoop operation
+  completes.
+* The sqoop process will trigger data integrity checks by issuing an API request
+  to [trigger the GitHub Action
+  workflow](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event).
+* As an initial MVP, data integrity results will print to the console in the
+  output of the GitHub Action workflow.
+* As an initial MVP, we will subscribe to [workflow
+  notifications](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/notifications-for-workflow-runs)
+  to get notified about results of data integrity checks.
+* As an iterative improvement, we will design a more complicated result
+  notification process using e.g. [stored dbt test
+  failures](https://docs.getdbt.com/reference/resource-configs/store_failures)
+  and [AWS SNS](https://aws.amazon.com/sns/) for notification management.
+
+### Pros
+
+* Designed around software engineering best practices (version control, reproducibility, testing, etc.)
+* Free and open source
+* Built on top of established, familiar tools (e.g. command line, SQL, Python)
+* Documentation and data validation out of the box
+
+### Cons
+
+* No native support for R scripting as a means of building models, so we would have to either rewrite our raw data extraction scripts or use some kind of hack like running our R scripts from a Python function
+* We would need to use a [community plugin](https://dbt-athena.github.io/) for Athena support; this plugin is not supported on dbt Cloud, if we ever decided to move to that
+* Requires a separate orchestrator for automation, monitoring, and alerting
+* Tests currently do not support the same rich documentation descriptions that other entities do (see [this GitHub issue](https://github.com/dbt-labs/dbt-core/issues/2578))
+
+### Open questions
+
+* Given the way tests are defined in dbt, we expect each validation check to be
```

Comment:
This might be a more substantial problem if we include [our CTAS](https://github.com/ccao-data/data-architecture/tree/master/aws-athena/ctas) in the dbt DAG. Some of these can take hours to run, so we may end up waiting for them to run sequentially and thus timeout the GH actions job limit.